azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/Utils.java

```
@@ -369,4 +371,53 @@ public FileVisitResult visitFileFailed(Path file, IOException exc) {
         }
         return tarFilePath;
     }
+
+    public static boolean isSpringBootProject(Path projectPath) {
+        try {
+            // Check if `pom.xml` exists and contains Spring Boot references
+            Path pomPath = projectPath.resolve("pom.xml");
+            if (Files.exists(pomPath) &&
+                (containsText(pomPath, "org.springframework.boot") ||
+                    containsText(pomPath, "spring-boot-maven-plugin"))) {
+                return true;
+            }
+
+            // Check if `build.gradle` exists and contains Spring Boot references
+            Path gradlePath = projectPath.resolve("build.gradle");
+            if (Files.exists(gradlePath) &&
+                (containsText(gradlePath, "org.springframework.boot") ||
+                    containsText(gradlePath, "spring-boot-starter"))) {
+                return true;
+            }
+
+            // Check for Spring Boot configuration files
+            Path resourcesPath = projectPath.resolve("src/main/resources");
+            if (Files.exists(resourcesPath)) {
+                if (Files.exists(resourcesPath.resolve("application.properties")) ||
+                    Files.exists(resourcesPath.resolve("application.yml"))) {
+                    return true;
+                }
+            }
+        } catch (IOException e) {
```

In which case will we meet the exception? Given that we had some check on existence, I think it would be better we output some error in this case to notify the customer about the exception, rather than fail silently and treat it as non-spring-boot app.